src/pages/index.js

```diff
@@ -1,62 +1,169 @@
 const profileEditBtn = document.querySelector("#profile-edit-btn");
+const profileImgEditBtn = document.querySelector("#edit-avatar-btn");
 // forms
 const editForm = document.querySelector("#profile-edit-modal");
 const addForm = document.querySelector("#profile-add-modal");
+const profileImageForm = document.querySelector("#edit-profile-img-modal");
 // form inputs
 const profileNameInput = editForm.querySelector("#profile-name-input");
 const profileDescriptionInput = editForm.querySelector(
   "#profile-description-input"
 );
 
-const userInfo = new UserInfo("#profile-name", "#profile-des");
+const userInfo = new UserInfo(
+  "#profile-name",
+  "#profile-des",
+  "#profile-image"
+);
+let userID;
+api
+  .getUserinfo()
+  .then((data) => {
+    const { name, about, avatar, _id } = data;
+    userID = _id;
+    userInfo.setUserInfo(name, about);
+    userInfo.setUserProfileImage(avatar);
+  })
+  .catch((error) => {
+    console.error("Error fetching data:", error);
+  });
 
 // functions
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profileEditPopup.close();
+  const { name, about } = data;
+  api
+    .updateProfile(name, about)
+    .then(() => {
+      profileEditPopup.editSubmitBtn();
+    })
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      profileEditPopup.close();
+      profileEditPopup.resetSubmitBtn();
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
 }
 
 function handleCardFormSubmit(data) {
-  const name = data.title;
-  const link = data.imageUrl;
-  const card = createCard({ name, link });
-  cardSection.prependItem(card);
-  profileAddPopup.close();
+  api.postCard(data).then((data) => {
+    const card = createCard(data);
+    cardSection.prependItem(card);
+    profileAddPopup.close();
+    profileAddPopup.resetSubmitBtn();
+  });
+}
+
+function handleDeleteIcon(cardId, card) {
+  deleteCardPopup.open(cardId);
+  deleteCardPopup.setSubmitAction(() => {
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        deleteCardPopup.close();
+        card.remove();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleProfileImgFormSubmit(data) {
+  api
+    .updateProfileImg(data.link)
+    .then(() => {
+      userInfo.setUserProfileImage(data.link);
+      profileImagePopup.close();
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
 }
 
 function handleCardClick(name, link) {
   previewImagePopup.open(name, link);
 }
+function addLike(cardId, likeCounter) {
+  api
+    .addLike(cardId)
+    .then((res) => {
+      likeCounter.textContent = res.likes.length;
+    })
+    .catch((error) => {
+      console.error("Error fetching data:", error);
+    });
+}
+
+function deleteLike(cardId, likeCounter) {
+  api
+    .deleteLike(cardId)
+    .then((res) => {
+      likeCounter.textContent = res.likes.length;
+    })
+    .catch((error) => {
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors